src/Mapping/ClassMetadata.php

```diff
@@ -582,9 +582,9 @@ public function getReflectionProperties(): array|LegacyReflectionFields
     }
 
     /**
-     * Gets the ReflectionProperties of the mapped class.
+     * Gets the PropertyAccessors of the mapped class.
      *
-     * @return PropertyAccessor[] An array of PropertyAccessor instances.
```

Comment:
This does not mean list, this means array. So this was correct. Therefore, this change should target 3.6.x